### PR TITLE
Add libnetwork agent mode support

### PR DESCRIPTION
**Pull request by mrjana**
libnetwork agent mode is a mode where libnetwork can act as a local
agent for network and discovery plumbing alone while the state
management is done elsewhere. This completes the support for making
libnetwork and its associated drivers to be completely independent of a
k/v store(if needed) and work purely based on the state information
passed along by some some external controller or manager. This does not
mean that libnetwork support for decentralized state management via a
k/v store is removed.

Signed-off-by: Jana Radhakrishnan <mrjana@docker.com>

### CODE REVIEW

**Comment by mavenugo** [comment]
@mrjana before starting the review. I just wanted to express my :clap:

**Comment by aboch** [comment]
LGTM

File: api/api.go [hunk 1]
```diff
@@ -693,10 +704,13 @@ func procAttachBackend(c libnetwork.NetworkController, vars map[string]string, b
 		setFctList = append(setFctList, libnetwork.CreateOptionAlias(name, alias))
 	}
 
+	log.Printf("Entering api join ...")
```
Comment (aboch) [inline api/api.go]: I am guessing these logs were added for debugging. If so you may want to remove them.

**Comment by mrjana** [comment]
@aboch Removed the unnecessary debug logs.

**Comment by sanimej** [comment]
LGTM

**Pull request merged**